src/common/libutil/dirwalk.c

```diff
@@ -233,7 +233,8 @@ static int dirwalk_traverse (dirwalk_t *d, dirwalk_filter_f fn, void *arg)
                 dirwalk_stop (d, errno);
             continue;
         }
-        if (S_ISDIR (d->current->sb.st_mode)) {
+        if (S_ISDIR (d->current->sb.st_mode)
+            && !(d->flags & DIRWALK_NORECURSE)) {
```

Comment:
There's no use case for this, so I hesitate to even mention it, but will this break the use of `DIRWALK_FIND_DIR` with `DIRWALK_NORECURSE`? We could just issue an error if both flags are used together, or I think if you just `dirwalk_visit()` the directory instead of `dirwalk_traverse()` when `DIRWALK_FIND_DIR` is requested, that might fix it? (after 2 minutes of examining code, so I might have got it wrong)

Reply:
Oh, thanks, that should have been obvious sorry!